### ISSUE TITLE
DataType Examples are not parsed

### ISSUE THREAD

**Issue by comikem** [mulesoft/api-designer#406]
In attempting to use dataTypes in place of schemas I have defined a type with an example.  The example in the type is ignored by the console or mocking service unless the example is explicitly defined in the response's specific media type.  

Example:
Type Definition:
```RAML
types:
  Member:
    properties:
      firstName: string
      lastName: string
      age: integer
    examples:
      Bob:
        value:
          firstName: "Bob"
          lastName: "Slidell"
          age: 42
      Bill:
        value:
          firstName: "Bill"
          lastName: "Lumberg"
          age: 41
````

If a response is resource type the example is not included
````RAML
resourceTypes:
  collection:
    usage: Use this resourceType to represent a collection of <<resourcePathName|!singularize>> items
    description: A collection of <<resourcePathName>>
    get:
      description: |
        Get all <<resourcePathName>>,
        optionally filtered
      responses:
        200:
          body:
            type: <<typeName|!singularize>>[]
````
Resource:
````RAML
/members:
  type:
     collection:
       typeName: <<resourcePathName|!uppercamelcase>>
  get:
````
This will show the Object type and its properties but not the example.  If the resource type explicitly defines the example it will be included in the console and returned by the Mock service.

````RAML
resourceTypes:
  collection:
    usage: Use this resourceType to represent a collection of <<resourcePathName|!singularize>> items
    description: A collection of <<resourcePathName>>
    get:
      description: |
        Get all <<resourcePathName>>,
        optionally filtered
      responses:
        200:
          body:
            type: <<typeName|!singularize>>[]
            application/json:            
              examples:
                Bob:
                  value:
                    firstName: "Bob"
                    lastName: "Slidell"
                    age: 42
                Bill:
                  value:
                    firstName: "Bill"
                    lastName: "Lumberg"
                    age: 41
````

The description of Examples in the 1.0 spec seems to indicate this would be a supported usage.  Full RAML below:
````RAML
#%RAML 1.0
title: TestRAML
baseUri: http://someapi.com/api
mediaType: application/json

resourceTypes:
  collection:
    usage: Use this resourceType to represent a collection of <<resourcePathName|!singularize>> items
    description: A collection of <<resourcePathName>>
    get:
      description: |
        Get all <<resourcePathName>>,
        optionally filtered
      responses:
        200:
          body:
            type: <<typeName|!singularize>>[]

types:
  Member:
    properties:
      firstName: string
      lastName: string
      age: integer
    examples:
      Bob:
        value:
          firstName: "Bob"
          lastName: "Slidell"
          age: 42
      Bill:
        value:
          firstName: "Bill"
          lastName: "Lumberg"
          age: 41
/members:
  type:
     collection:
       typeName: <<resourcePathName|!uppercamelcase>>
  get:
````

**Comment by sichvoge** [comment]
I can confirm that this seems to be a problem. Thanks for reporting, we will investigate that as soon as possible.

**Comment by sichvoge** [comment]
@comikem that will hopefully be solved with #402. Please have a bit patience with us as long as we resolve this issue ;)

**Comment by nmarinel01** [comment]
Fixed in develop branch
